airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/DatabaseConfigPersistence.java

```diff
@@ -1587,11 +1589,13 @@ private Set<String> getConnectorRepositoriesInUse(final DSLContext ctx) {
    *         users, and are not always the same as those in the seed.
    */
   @VisibleForTesting
-  Map<String, ConnectorInfo> getConnectorRepositoryToInfoMap(final DSLContext ctx) {
+  Map<String, ConnectorInfo> getConnectorRepositoryToInfoMap(final DSLContext ctx, final boolean runVersionCheckOnCustomConnectors) {
     return ctx.select(asterisk())
         .from(ACTOR_DEFINITION)
         .fetch()
         .stream()
+        .filter(
+            row -> runVersionCheckOnCustomConnectors || row.get(ACTOR_DEFINITION.RELEASE_STAGE) != ReleaseStage.custom)
```

Comment:
I don't think we even need to have this be configurable via a feature flag. Here's why:

[This](https://github.com/airbytehq/airbyte/blob/6c2cc80b1634805aef6efd74ca5a6837c79ae34b/airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/DatabaseConfigPersistence.java#L1771) is the logic that is checking the version of connectors.

For connectors that exist in the seed list, this makes sense to do because there may be a new version of an existing connector in the seed, which prompts us to update the connector definition in the database using this logic. Notice that this function receives `latestDefinitions` from the [seedConfigPersistence](https://github.com/airbytehq/airbyte/blob/6c2cc80b1634805aef6efd74ca5a6837c79ae34b/airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/DatabaseConfigPersistence.java#L1628). Therefore, there should **never** be a case where `latestDefinitions` contains a custom connector definition, because custom connectors are not in the seed.

Since custom connectors never have a new `latestDefinition`, there is no need to check for updates on them in this database initialization logic. Custom connectors will always require manual version updates, which makes sense.

Therefore, I don't think we need an environment variable to configure this here. I think we can just go ahead and add
```
.filter(row.get(ACTOR_DEFINITION.RELEASE_STAGE) != ReleaseStage.custom)
```

Reply:
That makes senses, I will add a comment. and remove the feature flag.